fabric-api-base/src/main/java/net/fabricmc/fabric/api/event/EventFactory.java

```diff
@@ -70,9 +77,16 @@ public static void invalidate() {
 	 * @param <T>            The listener type.
 	 * @return The Event instance.
 	 */
-	// TODO: Deprecate this once we have working codegen
 	public static <T> Event<T> createArrayBacked(Class<T> type, T emptyInvoker, Function<T[], T> invokerFactory) {
-		return EventFactoryImpl.createArrayBacked(type, emptyInvoker, invokerFactory);
+		return createArrayBacked(type, invokers -> {
+			if (invokers.length == 0) {
+				return emptyInvoker;
+			} else if (invokers.length == 1) {
```

Comment:
You should remove this branch.

Reply:
It it is a bit faster as it bypasses the usual loop in the factory.

Reply:
I mean, the empty invoker and the factory can be customized to do extra work beyond calling the invokers directly. This branch destroys the possibility of customizing in the case when `listeners.length == 1`.

Reply:
Also the lambda param should be better named `listeners`, as invoker is a single object that dispatch events in the fabric base api definition.

Reply:
That's true, but this will be a bit faster for factories that only deal with listener composition, so most of them, and certainly all the existing ones. Factories that want to add custom behavior can use the other overload.

Reply:
> Also the lambda param should be better named listeners, as invoker is a single object that dispatch events in the fabric base api definition.

Yup, was distracted apparently.

Reply:
> 
> 
> That's true, but this will be a bit faster for factories that only deal with listener composition, so most of them, and certainly all the existing ones. Factories that want to add custom behavior can use the other overload.

Imo you should document that for invoker customization purposes, this shouldn't be used if it aims to do more than just calling invokers.

Say the event in your pr description is seen by others. Seeing that the empty invoker version is more "optimized", they incorrectly optimize the example event `SLOT_CLICK` in the pr description, only to see it go :boom: because a custom invoker ain't created when there is only one listener.

Reply:
Or should we change the empty invoker variant to have an alternative name? Like `createDirectArrayBacked` (bad name here) to indicate this aims to directly dispatch events?

Reply:
It is painful to come up with a sufficiently distinct name for such a tiny behavior difference, and we would need to essentially duplicate the function and deprecate the old one just for the rename. I'm not fond of it.